qiskit/transpiler/coupling.py

```diff
@@ -403,6 +391,45 @@ def largest_connected_component(self):
         """Return a set of qubits in the largest connected component."""
         return max(rx.weakly_connected_components(self.graph), key=len)
 
+    def get_component_subgraphs(self, respect_direction: bool = False) -> List["CouplingMap"]:
```

Comment:
Do we have a feel for whether this is something we'll need to access enough times in different contexts that it makes sense to keep as a cached property on `CouplingMap`?

Reply:
In the preset pass managers I think the worst case in practice is we'd call this method twice once for a heuristic layout and once for routing. I don't think anything else needs to worry about this because after routing nothing should need to think about the individual components.

Reply:
Yeah, that's probably fair.  If we're playing naming nitpicks, I'd vote in favour of shortening it to `CouplingMap.components()` or something, but I don't care a great deal.

Reply:
I don't actually know what the scaling performance is for the connected components functions in rustworkx are off the top of my head. I don't expect it to be a performance bottleneck, but it doesn't hurt to cache it either, I'll add that and also rename the methods to `.components()`

Reply:
Done in: https://github.com/Qiskit/qiskit-terra/pull/9710/commits/25d3379ffe4860433c33476907d41195fafbd0f6